File: Chain of Responsibility/PHP/Request.php
<?php

class Request {
    private bool $authenticated;
    private bool $authorized;
    private string $route;

    function __construct(bool $authenticated, bool $authorized, string $route) {
        $this->authenticated = $authenticated;
        $this->authorized = $authorized;
        $this->route = $route;
    }

    function is_authenticated(){
        return $this->authenticated;
    }

    function is_authorized(){
        return $this->authorized;
    }

    function get_route() {
        return $this->route;
    }
}

File: Chain of Responsibility/Dart/Request.dart
class Request {
  bool? _authenticated;
  bool? _authorized;
  String? _route;

  Request(this._authenticated, this._authorized, this._route);

  bool? get is_authenticated => _authenticated;
  bool? get is_authorized => _authorized;
  String? get get_route => _route;
}
